pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -550,6 +559,70 @@ class CollectionArtifactDownloadView(views.APIView):
 
     DEFAULT_ACCESS_POLICY = _PERMISSIVE_ACCESS_POLICY
 
+    @staticmethod
+    def log_download(request, filename, distro_base_path):
+        """Log the download of the collection version."""
+
+        def _get_client_ip(request):
+            x_forwarded_for = request.META.get("HTTP_X_FORWARDED_FOR")
+            if x_forwarded_for:
+                ip = x_forwarded_for.split(",")[0]
+            else:
+                ip = request.META.get("REMOTE_ADDR")
+            return ip
+
+        def _get_user_agent(request):
+            user_agent = request.headers.get("user-agent")
+            if not user_agent:
+                return "unknown"
+
+            return user_agent
+
+        def _get_identity(request):
+            if not isinstance(request.auth, dict):
+                return None
+
+            x_rh_identity = request.auth.get("rh_identity")
+            if not x_rh_identity:
+                return None
+
+            return x_rh_identity["identity"]
+
+        def _get_user(request):
+            identity = _get_identity(request)
+            if (not identity) or (not identity.get("user")):
+                return None
+
+            try:
+                return get_user_model().objects.get(id=identity["user"]["user_id"])
+            except ObjectDoesNotExist:
+                return None
+
+        def _get_org_id(request):
+            identity = _get_identity(request)
+            if (not identity) or (not identity.get("internal")):
+                return None
+
+            return identity["internal"]["org_id"]
+
+        def _get_collection_from_filename(filename):
+            collection_namespace = filename.split("-")[-3]
+            collection_name = filename.split("-")[-2]
+            collection_version = filename.split("-")[-1][:5]
+
+            return CollectionVersion.objects.get(
+                namespace=collection_namespace, name=collection_name, version=collection_version
+            )
+
+        CollectionDownloadLog.objects.create(
+            collection_version=_get_collection_from_filename(filename),
+            user=_get_user(request),
+            ip=_get_client_ip(request),
+            org_id=_get_org_id(request),
+            user_agent=_get_user_agent(request),
+            repository=AnsibleDistribution.objects.get(name=distro_base_path).repository,
```

Comment:
use `get_object_or_404` instead of `objects.get` so this doesn't throw a 500 if the object doesn't exist.